pymc_marketing/clv/distributions.py

```diff
@@ -609,28 +608,23 @@ def logp(value, alpha, beta, gamma, delta, T):
             t_x, alpha, beta, gamma, delta, T
         )
 
-        def logp_customer_died(t_x_i, x_i, alpha_i, beta_i, gamma_i, delta_i, T_i):
-            i = pt.scalar("i", dtype=int)
-            died = pt.lt(t_x_i + i, T_i)
+        i = pt.scalar("i", dtype=int)
+        died = pt.lt(t_x + i, T)
 
-            unnorm_logprob_customer_died_at_tx_plus_i = betaln(
-                alpha_i + x_i, beta_i + t_x_i - x_i + i
-            ) + betaln(gamma_i + died, delta_i + t_x_i + i)
-
-            # Maximum prevents invalid T - t_x values from crashing logp
-            i_vec = pt.arange(pt.maximum(T_i - t_x_i, 0) + 1)
-            unnorm_logprob_customer_died_at_tx_plus_i_vec = vectorize_graph(
-                unnorm_logprob_customer_died_at_tx_plus_i, replace={i: i_vec}
-            )
-
-            return pt.logsumexp(unnorm_logprob_customer_died_at_tx_plus_i_vec)
+        unnorm_logp_died_at_tx_plus_i = betaln(alpha + x, beta + t_x - x + i) + betaln(
+            gamma + died, delta + t_x + i
+        )
 
-        unnorm_logp, _ = scan(
-            fn=logp_customer_died,
-            outputs_info=[None],
-            sequences=[t_x, x, alpha, beta, gamma, delta, T],
+        # Maximum prevents invalid T - t_x values from crashing logp
+        max_range = pt.maximum(pt.max(T - t_x), 0)
```

Comment:
Actually I think here might be the issue as this maximum varies over “i” in the scan implementation and therefore this global maximum will not match 🤔

Reply:
And not I think I got the reason of pt.ge(t_x, i) from the unnorm_logp_died_at_tx_plus_i on the first place 🙈 🤦

Reply:
`pt.maximum` seems redundant to me because `T` is the same value across all customers for this model; it'll never be less than 0. Maybe add/subtract 1 from `max_range` so that it indexes properly?

Reply:
The maximum is just a sanity check. For constant data it will be constant folded (evaled once and remove from the graph)

Reply:
> And not I think I got the reason of pt.ge(t_x, i) from the unnorm_logp_died_at_tx_plus_i on the first place 🙈 🤦

I don't understand this comment

Reply:
The formula is slightly different after the subject "died", in that case it needs a +1 in one of the terms, that what it was doing

Reply:
So about the comment above: "I was just saying I understood the reason of the `np.where` and I think I fixed in https://github.com/pymc-labs/pymc-marketing/pull/707/commits/da6568c846858723cc31ea931e51f3a6c0d7c375